scripts/model_explorer.py

```diff
@@ -0,0 +1,247 @@
+        self.runtime_combobox = runtime_combobox
+
+        reload_button = ttk.Button(mainframe, text ="Reload", command=self.reload).grid(row=0, column=2)
+        self.reload_button = reload_button
+
+        # dropdown
+        dropdown_var = tkinter.StringVar(root)
+        # Dictionary with options
+        choices = list([str(i) for i in range(len(self.filenames))])
+        dropdown_var.set('0') # set the default option
+        label = ttk.Label(mainframe, text="Filenames").grid(row=1, column=1)
+        dropdown_var.trace('w', self.change_dropdown)
+        self.dropdown_var = dropdown_var
+
+        # note that width is not automatically adjusted
+        combobox = ttk.Combobox(mainframe, textvariable=dropdown_var, values=choices, width=90)
+        combobox.grid(row=2, column=1)
+        self.combobox = combobox
+
+        next_button = ttk.Button(mainframe, text ="Next", command=self.next).grid(row=2, column=2)
+        prev_button = ttk.Button(mainframe, text ="Prev", command=self.prev).grid(row=2, column=0)
+        self.next_button = next_button
+        self.prev_button = prev_button
+
+        self.label = label
+
+        # setup figure
+        figure = plt.figure()
+        figure.tight_layout()
+        self.figure = figure
+
+        # plotter
+        canvas = FigureCanvasTkAgg(self.figure, master=root)  # A tk.DrawingArea.
+        canvas.draw()
+        canvas.get_tk_widget().pack(side=tkinter.TOP, fill=tkinter.BOTH, expand=1)
+
+        toolbar = NavigationToolbar2Tk(canvas, root)
+        toolbar.update()
+        canvas.get_tk_widget().pack(side=tkinter.TOP, fill=tkinter.BOTH, expand=1)
+        self.canvas = canvas
+        self.toolbar = toolbar
+
+        # TODO: add ui for additional args
+        self.additional_args = dict(
+            score_threshold=0.9,
+            iou_threshold=0.1,
+        )
+    
+    def quit(self):
+        """quit gui program
+        """        
+        self.root.quit()     # stops mainloop
+        self.root.destroy()  # this is necessary on Windows to prevent
+    
+    def load_model(self):
+        """prompt filename dialog and then load model
+        """
+        filename = filedialog.askopenfilename()
+        if filename:
+            self.model_path = filename
+            self.reload()
+    
+    def reload(self):
+        """load model from saved model_path & runtime
+        """
+        if Path(self.model_path).exists() :
+            kwargs = dict(
+                model_path=self.model_path,
+                runtime=self.runtime_var.get(),
+            )
+            print('loading model, this may take a while')
+            self.model = InferenceHelper.create_runtime_model(**kwargs)
+            print('loading model done:', type(self.model.model))
+            self.runtime = []
+            title = f"{self.title}: {self.model_path}"
+            self.root.title(title)
+        else:
+            print(f'{filename} doesnt exists')
```

Comment:
`filename` is not defined here, where is this variable come from?

Reply:
fixed